package.json

```diff
@@ -41,6 +41,7 @@
     "gatsby": "4.1.0",
     "gatsby-background-image": "^1.5.3",
     "gatsby-image": "^3.11.0",
+    "gatsby-plugin-breadcrumb": "^12.2.0",
```

Comment:
I have mixed feelings about this plugin:
- it only has 51 stars on Github, so not many people seem to use it
- it adds black-box behavior to our build. I don't understand where the breadcrumbs come from or how they are generated. Breadcrumbs seem to be a very easy concept: An array of links that is shown as list. Wouldn't it be simpler to just manually add a `breadcrumbs` array to the `pageContext` of each page?

Reply:
Not sure about the 51 stars and how much that says about the quality of the plugin. On gatsbyjs.com it has 12.9k downloads (not sure whether this is overall or yearly or monthly or hourly...). It seems to be actively maintained and apparently works well for us, so why not use it?

Generally I do very much like your second point, asking whether we can do the same without using such an external black-box dependency. Looking at the code from that plugin I'm not sure if it's actually that easy, though. If I understand you correctly you suggest to basically hardcode the "array of links" to each page... I'd like to avoid that if we can instead have such a plugin to take care of it. What are the downsides?

Reply:
Better late than never: The plugin does a lot of things we don't need (like click tracking to get the breadcrumb path). What I don't like is when a plugin adds something to the page context. This makes it later very hard to understand where those page context variables come from, as you just see them popping up in the templates. Also, this plugin doesn't work for all pages but just for some pages we have (as the career pages are added via the Personio API). 

In general, I prefer to have all link logic either in `gatsby-node`, where you can see how links are generated. Or directly from an API. This way you have a better overview of how those components get data.